vsintegration/src/FSharp.Editor/Commands/XmlDocCommandService.fs

```diff
@@ -24,12 +24,12 @@ type internal XmlDocCommandFilter
         workspace: VisualStudioWorkspace
      ) =
 
-    let document =
-        // There may be multiple documents with the same file path.
-        // However, for the purpose of generating XmlDoc comments, it is ok to keep only the first document.
-        lazy(match workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath) |> Seq.toList with
-             | [] -> None
-             | documentId :: _ -> Some (workspace.CurrentSolution.GetDocument documentId))
+    let getLastDocument () =
+        // There may be multiple documents with the same file path, due to versioning .
+        // However, for the purpose of generating XmlDoc comments, it is ok to keep only the last document, since they're versioned.
+        match workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath) |> Seq.toList |> List.rev with
```

Comment:
Is it guaranteed to be sorted by version here or should we sort it to be safe?

Reply:
> Is it guaranteed to be sorted by version here or should we sort it to be safe?

There's nothing in the docs about it, but we have no better way as for now (before we migrate to autocomplete/snippet-like docs generation). It works now (gives correct document), might stop working in the future thought. It fixes the issue for now.